www/js/services.js

```diff
@@ -102,6 +102,91 @@ angular.module('emission.services', [])
                "Email cancel reported, seems to be an error on android");
         });
     }
+
+    this.dataCollectionSetConfig = function(config) {
+      return new Promise(function(resolve, reject) {
+          resolve(window.cordova.plugins.BEMDataCollection.setConfig(config));
+      });
+    };
+
+    this.dataCollectionGetConfig = function() {
+      return new Promise(function(reslove, reject) {
+          reslove(window.cordova.plugins.BEMDataCollection.getConfig());
+      });
+    };
+    this.serverSyncSetConfig = function(config) {
+      return new Promise(function(resolve, reject) {
+          resolve(window.cordova.plugins.BEMServerSync.setConfig(config));
+      });
+    };
+
+    this.serverSyncGetConfig = function() {
+      return new Promise(function(reslove, reject) {
+          reslove(window.cordova.plugins.BEMServerSync.getConfig());
+      });
+    };
+
+    this.getAccuracyOptions = function() {
+      return new Promise(function(reslove, reject) {
+          reslove(window.cordova.plugins.BEMDataCollection.getAccuracyOptions());
+      });
+    };
+
+    this.getUserEmail = function() {
+      return new Promise(function(reslove, reject) {
+        window.cordova.plugins.BEMJWTAuth.getUserEmail(function(response) {
+            reslove(response);
+        }, function(error) {
+            reject(error);
+        });
+      });
+    };
+
+    this.getState = function() {
+      return new Promise(function(reslove, reject) {
+        window.cordova.plugins.BEMDataCollection.getState(function(response) {
```

Comment:
Everything in `datacollection.js` is already a promise except for this
function (that I probably forgot).

You can use `getAccuracyOptions` right above it as a template for how to
convert it - basically, the `resolve` function is the `successCallback` and
the `reject` function is the `errorCallback`

Shankari